internal/alb/tg/targetgroup.go

```diff
@@ -150,6 +156,33 @@ func (controller *defaultController) reconcileTGInstance(ctx context.Context, in
 	return instance, nil
 }
 
+// resolveServiceHealthCheckPort checks if the service-port annotation is a string. If so, it tries to look up a port with the same name
+// on the service and use that port's NodePort as the health check port.
+func resolveServiceHealthCheckPort(store store.Storer, namespace string, serviceName string, servicePortAnnotation intstr.IntOrString) (* string) {
+	servicePort := servicePortAnnotation.String()
+
+	//If the annotation uses the default port ("traffic-port"), do not try to look up a port by that name.
+	if servicePortAnnotation.Type == intstr.String && servicePortAnnotation.StrVal != healthcheck.DefaultPort {
```

Comment:
1. i think we can create an utility function under internal/k8s package.(which can be leveraged by https://github.com/kubernetes-sigs/aws-alb-ingress-controller/blob/aeb86a388d1e565cc6629a2b1d23fba3dcbdbc47/internal/ingress/backend/endpoint.go#L114 as well)
Something like 
```lookupServicePort(service corev1.Service, port intstr.IntOrString) (corev1.ServicePort error)```sounds good to me =.=(don't make store part of it)